### PR TITLE
Fix scanpy by using profile

### PR DESCRIPTION
**Pull request by mvdbeek**
numba write to the code directory or the home directory, neither of which is or should be writable. The profile version ensures this is written to an isolated home_dir for the job. 

FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by mvdbeek** [comment]
Huh, containerized test passed while the Conda test fails with
```
| Dataset Job Standard Error:
|  Traceback (most recent call last):
|  File "/tmp/tmppX0FjK/job_working_directory/000/2/tmpcO_7Mi", line 21, in <module>
|  copy=False)
|  File "/home/travis/miniconda3/envs/mulled-v1-141253bb58fad23ac50749589168696eca3ed3ab1e99608dbf718dbfa9556234/lib/python3.6/site-packages/scanpy/tools/_louvain.py", line 120, in louvain
|  import louvain
|  ModuleNotFoundError: No module named 'louvain'
```

**Comment by mvdbeek** [comment]
OK, louvain has actually been removed at https://github.com/bioconda/bioconda-recipes/commit/8cbb444c8e2c9ce073ba7ad023f79d0ae0693ee4#diff-42a29748c004d0dde077e3b41b022f3c, but the container wasn't rebuilt. So I guess we should add louvain as a separate requirement.

**Comment by bgruening** [comment]
Mh, the container is here:  1.4.4.post1--py_1 ... https://quay.io/repository/biocontainers/scanpy?tab=tags  isn't it?

**Comment by mvdbeek** [comment]
Hmm, the container seems to contain louvain, which it shouldn't according to the recipe ...:
```
Unable to find image 'quay.io/biocontainers/scanpy:1.4.4.post1--py_1' locally
1.4.4.post1--py_1: Pulling from biocontainers/scanpy
a3ed95caeb02: Already exists
b0dc45cd432d: Already exists
9466b3513669: Already exists
ddd482ea7b54: Already exists
4d69f833b9d8: Already exists
e7c454e5167d: Already exists
e38092b005c0: Already exists
f879b42dfe2b: Already exists
bc6d8d5a56f8: Pull complete
Digest: sha256:4cd4cd9a9c555c8178464ba0b69748938459009958d743dbdd1e100436b4fb0d
Status: Downloaded newer image for quay.io/biocontainers/scanpy:1.4.4.post1--py_1
bash-4.2# python
Python 3.7.3 | packaged by conda-forge | (default, Jul  1 2019, 21:52:21)
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import louvain
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'louvain'
>>> import scanpy as sc
>>> sc.tl.louvain
<function louvain at 0x7fe4b2579730>
>>> sc.tl.louvain()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: louvain() missing 1 required positional argument: 'adata'
>>> sc.tl.louvain('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/site-packages/scanpy/tools/_louvain.py", line 95, in louvain
    if adjacency is None and 'neighbors' not in adata.uns:
AttributeError: 'str' object has no attribute 'uns'
```

**Comment by bebatut** [comment]
Yep we will need louvain for sure.
I have also some local changes to the wrappers and was planning also to update to the latest version. I will wait for your changes to add mine

**Comment by mvdbeek** [comment]
This is ready for review

**Comment by bgruening** [comment]
@bebatut do you want to add your changes to this branch?

**Comment by mvdbeek** [comment]
Let's try to keep the changes frequent and small, this will help with debugging and reviewing.

File: tools/scanpy/macros.xml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <macros>
-    <token name="@version@">1.4.4</token>
-    <token name="@galaxy_version@"><![CDATA[@version@+galaxy0]]></token>
+    <token name="@version@">1.4.4.post1</token>
+    <token name="@profile@">19.01</token>
+    <token name="@galaxy_version@"><![CDATA[@version@+galaxy1]]></token>
```
Comment (bebatut) [inline tools/scanpy/macros.xml]: Why did you bump the galaxy version as the tool version already changed?

**Comment by bebatut** [comment]
I will add the changes in another PR as they are not ready

File: tools/scanpy/macros.xml (hunk 1 above)
Comment (mvdbeek) [reply tools/scanpy/macros.xml]: Ah ... last minute changes, I'll rebase this out. Thanks @bebatut

File: tools/scanpy/cluster_reduce_dimension.xml [hunk 2]
```diff
@@ -58,7 +58,9 @@ This data is accessible using the inspect tool for AnnData
     use_highly_variable=$method.use_highly_variable
 ]]></token>
     </macros>
-    <expand macro="requirements"/>
+    <expand macro="requirements">
+        <requirement type="package" version="0.6.1">louvain</requirement>
```
Comment (bernt-matthias) [inline tools/scanpy/cluster_reduce_dimension.xml]: This should be added to the conda recipe.

Comment (mvdbeek) [reply tools/scanpy/cluster_reduce_dimension.xml]: No, this has been removed from the conda recipe because it is an optional dependency.

Comment (bernt-matthias) [reply tools/scanpy/cluster_reduce_dimension.xml]: OK.

**Pull request merged**